src/main/java/edu/iu/terracotta/service/app/impl/OutcomeScoreServiceImpl.java

```diff
@@ -0,0 +1,80 @@
+package edu.iu.terracotta.service.app.impl;
+
+import edu.iu.terracotta.exceptions.DataServiceException;
+import edu.iu.terracotta.model.app.Outcome;
+import edu.iu.terracotta.model.app.OutcomeScore;
+import edu.iu.terracotta.model.app.Participant;
+import edu.iu.terracotta.model.app.dto.OutcomeScoreDto;
+import edu.iu.terracotta.repository.AllRepositories;
+import edu.iu.terracotta.service.app.OutcomeScoreService;
+import edu.iu.terracotta.service.app.OutcomeService;
+import edu.iu.terracotta.service.app.ParticipantService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class OutcomeScoreServiceImpl implements OutcomeScoreService {
+
+    @Autowired
+    AllRepositories allRepositories;
+
+    @Autowired
+    OutcomeService outcomeService;
+
+    @Autowired
+    ParticipantService participantService;
+
+    @Override
+    public List<OutcomeScore> findAllByOutcomeId(Long outcomeId) { return allRepositories.outcomeScoreRepository.findByOutcome_OutcomeId(outcomeId);}
+
+    @Override
+    public OutcomeScoreDto toDto(OutcomeScore outcomeScore){
+        OutcomeScoreDto outcomeScoreDto = new OutcomeScoreDto();
+        outcomeScoreDto.setOutcomeScoreId(outcomeScore.getOutcomeScoreId());
+        outcomeScoreDto.setOutcomeId(outcomeScore.getOutcome().getOutcomeId());
+        outcomeScoreDto.setParticipantId(outcomeScore.getParticipant().getParticipantId());
+        outcomeScoreDto.setScoreNumeric(outcomeScore.getScoreNumeric());
+
+        return outcomeScoreDto;
+    }
+
+    @Override
+    public OutcomeScore fromDto(OutcomeScoreDto outcomeScoreDto) throws DataServiceException {
+        OutcomeScore outcomeScore = new OutcomeScore();
+        outcomeScore.setOutcomeScoreId(outcomeScoreDto.getOutcomeScoreId());
+        outcomeScore.setScoreNumeric(outcomeScoreDto.getScoreNumeric());
+        Optional<Outcome> outcome =  outcomeService.findById(outcomeScoreDto.getOutcomeId());
```

Comment:
Have we checked that this outcome belongs to the exposure?  We should if not done before

Reply:
And surely before doing other things... so we save time if it is not allowed. So move this to the first line in the method

Reply:
As for checking if the outcome belongs to an exposure, that is done first thing in the Controller endpoints. The experiment is checked, the exposure is checked, and then the outcome is checked (or outcome and outcome score for the outcome score controller). Since this is the case I'm not sure that the order matters in the method.